### PR TITLE
AC-551: Implement Dark Mode (Experimental) , use CardView instead of custom Card and follow Material guidelines for applying themes.

### PR DESCRIPTION
**Pull request by deepak140596**
## Description of what I changed

- Added option to switch onto Dark Mode
- Removed unnecessary use of @styles/AppTheme
- Made use of @styles/AppThemeOrig in most of the Activities (it provides default Action Bar)
- Removed unnecessary use of @drawable/card layout as it was incompatible with current material designs
- Removed unnecessary use of setTextColor, setBackgroundColor, setBackgroundTheme, setBackground

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-551

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/res/layout/list_visit_group.xml [hunk 1]
```diff
@@ -12,55 +12,57 @@
   ~ Copyright (C) OpenMRS, LLC.  All Rights Reserved.
   -->
 
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/layout/list_visit_group.xml]: @deepak140596 What is the change here? Any screenshot of the improved list view?

File: openmrs-client/src/main/res/layout/list_vital_group.xml [hunk 2]
```diff
@@ -13,55 +12,61 @@
   ~ Copyright (C) OpenMRS, LLC.  All Rights Reserved.
   -->
 
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/layout/list_vital_group.xml]: Show us a screenshot of this improved list view.

File: openmrs-client/src/main/res/values/color.xml [hunk 3]
```diff
@@ -26,6 +26,7 @@
     <color name="dark_teal">#00463f</color>
     <color name="primary">#00463f</color>
     <color name="primary_dark">#01302b</color>
+    <color name="colorAccent">#FF4081</color>
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/values/color.xml]: Follow the naming convention. Shouldn't be in camel case.

File: openmrs-client/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -46,5 +46,6 @@
     <dimen name="custom_fab_bottom_margin_155">155dp</dimen>
     <dimen name="custom_fab_standard_21">21dp</dimen>
     <dimen name="padding_16">16dp</dimen>
+    <dimen name="margin_4">4dp</dimen>
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/values/dimens.xml]: Replace with `margin_small`

File: openmrs-client/src/main/res/layout/list_visit_group.xml (hunk 1 above)
Comment (deepak140596) [reply openmrs-client/src/main/res/layout/list_visit_group.xml]: Thanks @f4ww4z . I am posting the UI. All the rows are now using CardViews with ripple effect for onClick.

**Comment by deepak140596** [comment]
These are changed UI and for contrast, both the Day and Night UIs are posted.

Capture+_2019-07-15-17-34-03
Capture+_2019-07-15-17-35-22
Capture+_2019-07-15-17-36-06
Capture+_2019-07-15-17-36-15
Capture+_2019-07-15-17-36-42
Capture+_2019-07-15-17-36-58
Capture+_2019-07-15-17-38-19
Capture+_2019-07-15-17-38-37
Capture+_2019-07-15-17-39-05
Capture+_2019-07-15-17-39-11
Capture+_2019-07-15-17-39-17
Capture+_2019-07-15-17-39-23
Capture+_2019-07-15-17-39-36
Capture+_2019-07-15-17-39-50
Capture+_2019-07-15-17-40-02

File: openmrs-client/src/main/res/layout/list_visit_group.xml (hunk 1 above)
Comment (deepak140596) [reply openmrs-client/src/main/res/layout/list_visit_group.xml]: One issue still remains, we need to migrate the PNG/JPG icons to vectors. I will create a new issue for that.

Comment (f4ww4z) [reply openmrs-client/src/main/res/layout/list_visit_group.xml]: > One issue still remains, we need to migrate the PNG/JPG icons to vectors. I will create a new issue for that.

Noted, ping me there.

**Comment by f4ww4z** [comment]
@deepak140596 You'll need to implement dark mode in 'Add Patients' screen of 'Find Patients' section. See the screenshot:
image

Also, try **full white** (#FFFFFF) for all text color when in dark mode, I think [it will look better](https://material.io/design/color/dark-theme.html#behavior).

**Comment by deepak140596** [comment]
@f4ww4z , I haven't used any hardcoded text color. Android is deciding the text color. We have two options for setting text colors: i) not using any attributes for text color ii) using text color as
            android:textColor="?android:attr/textColorPrimary".
The second option will make the text full black (#000000) in Light Mode and full white (#ffffff) in Dark Mode. This needs to be specifically set and should be used in headings. 

For body we should just leave it as it is. In the same example we can see there are 2 shades of white.

One exception though, I have used a neutral green color for "Unable to login" option to highlight. Should I replace the color to White?

**Pull request merged**